src/filesystem/wal_writer_file_lease.h

```diff
@@ -1,20 +1,45 @@
 // Copyright (c) 2016 Alexander Gallego. All rights reserved.
 //
 #pragma once
-#include <core/future.hh>
+#include <core/fstream.hh>
+#include <core/sstring.hh>
 
 namespace smf {
-/// \brief exposes what the \ref wal_writer_node.h
-/// needs which is effecitvely a seastar::output_stream<char>
-/// however, we want to prevent the readers and writers to be accessing the
-/// same files
+
+/// \brief changes the filename while it's being written to:
+///   "locked:" + filename
+/// upon successful closing of the stream, it will rename it to the correct
+/// name. It is also open w/ exclusive acess. Any other thread accessing the
+/// file will fail/throw. Caller must ensure that only one filename is
+/// needed for writing at a time
+/// see \ref wal_file_manager.h
+///
+/// Rejected Solutions: Keep a pointer to the file_manager.
+///
+/// having a pointer to the file manager introduces unnecessary complexity,
+/// given that we have to monitor already the filesystem for files dropped
+/// by other cores etc.
+///
+/// The tradeoff here is that since we change the filename, we rely on the
+/// filesystem manger to "heal" the files. That is during startup,
+/// rename all the files that start w/ locked: to remove the prefix.
```

Comment:
I'll add that. 

but effectively it cannot start unless the folder (my view of the filesystem) is correctly healed ready to go. 

It will throw and abort.